### ISSUE TITLE
Opening brace of newline should be on same line as condition in `if let`

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#1690]
```
if let foo = bar(
    dfadfas,
    sdafasfas,
) {
    ...
}
```

In some cases, we put a newline between `) {`

**Comment by topecongiro** [comment]
Does this only apply to `if let`, or to other control flows as well?

**Comment by nrc** [comment]
It should apply to all control flow statements, I've only seen the error with `if let`

**Comment by topecongiro** [comment]
@nrc Do you think this issue can be closed via #1711? Are there other cases where the opening brace should be on the same line?

**Comment by nrc** [comment]
Yes. We can open new issues if we find them.
